packages/core/contracts/common/implementation/FixedPoint.sol

```diff
@@ -373,4 +377,386 @@ library FixedPoint {
+            return Signed(mulTowardsZero.add(valueToAdd));
+        } else {
+            return Signed(mulTowardsZero);
+        }
+    }
+
+    /**
+     * @notice Multiplies an `Signed` and an unscaled int256 and "ceil's" the product, reverting on overflow.
+     * @param a a FixedPoint.Signed.
+     * @param b a FixedPoint.Signed.
+     * @return the product of `a` and `b`.
+     */
+    function mulCeil(Signed memory a, int256 b) internal pure returns (Signed memory) {
+        // Since b is an int, there is no risk of truncation and we can just mul it normally
+        return Signed(a.rawValue.mul(b));
+    }
+
+    /**
+     * @notice Divides one `Signed` by an `Signed`, reverting on overflow or division by 0.
+     * @dev This will "floor" the quotient.
+     * @param a a FixedPoint numerator.
+     * @param b a FixedPoint denominator.
+     * @return the quotient of `a` divided by `b`.
+     */
+    function div(Signed memory a, Signed memory b) internal pure returns (Signed memory) {
+        // There are two caveats with this computation:
+        // 1. Max value for the number dividend `a` represents is ~10^41, otherwise an intermediate value overflows.
+        // 10^41 is stored internally as an int256 10^59.
+        // 2. Results that can't be represented exactly are truncated not rounded. E.g., 2 / 3 = 0.6 repeating, which
+        // would round to 0.666666666666666667, but this computation produces the result 0.666666666666666666.
+        return Signed(a.rawValue.mul(SFP_SCALING_FACTOR).div(b.rawValue));
+    }
+
+    /**
+     * @notice Divides one `Signed` by an unscaled int256, reverting on overflow or division by 0.
+     * @dev This will "floor" the quotient.
+     * @param a a FixedPoint numerator.
+     * @param b an int256 denominator.
+     * @return the quotient of `a` divided by `b`.
+     */
+    function div(Signed memory a, int256 b) internal pure returns (Signed memory) {
+        return Signed(a.rawValue.div(b));
+    }
+
+    /**
+     * @notice Divides one unscaled int256 by an `Signed`, reverting on overflow or division by 0.
+     * @dev This will "floor" the quotient.
+     * @param a an int256 numerator.
+     * @param b a FixedPoint denominator.
+     * @return the quotient of `a` divided by `b`.
+     */
+    function div(int256 a, Signed memory b) internal pure returns (Signed memory) {
+        return div(fromUnscaledInt(a), b);
+    }
+
+    /**
+     * @notice Divides one `Signed` by an `Signed` and "ceil's" the quotient, reverting on overflow or division by 0.
+     * @param a a FixedPoint numerator.
+     * @param b a FixedPoint denominator.
+     * @return the quotient of `a` divided by `b`.
+     */
+    function divAwayFromZero(Signed memory a, Signed memory b) internal pure returns (Signed memory) {
+        int256 aScaled = a.rawValue.mul(SFP_SCALING_FACTOR);
+        int256 divTowardsZero = aScaled.div(b.rawValue);
+        // Manual mod because SignedSafeMath doesn't support it.
```

Comment:
any reason why they don't support it?

Reply:
Not sure. Probably because mod is a little unintuitive (it takes the sign of the first arg in solidity, but some languages differ) when done with negative numbers. However, comparing against 0 still has the properties we care about AFAIK.